src/collectors/loggly/loggly.py

```diff
@@ -0,0 +1,246 @@
+
+            set_defaults(config)
+
+            config.update({
+                'search_URI': 'https://{1.subdomain}.loggly.com'
+                              '/apiv2/search?q={0.query}&from='
+                              '-{1.interval}s&until=now'.format(
+                                  config, self.config_bunch),
+                'events_URI': 'https://{0.subdomain}.loggly.com'
+                              '/apiv2/events?rsid=%s'.format(
+                                  self.config_bunch)})
+
+    def collect(self):
+        """Collect metrics from loggly."""
+        def parse_timestamp(metric_config, timestamp):
+            if metric_config.timestamp_format is 'int':
+                return datetime.datetime.fromtimestamp(
+                    timestamp / 10**(ceil(log10(timestamp)) - 10))
+            else:
+                return datetime.datetime.strptime(timestamp,
+                                                  self.config_bunch
+                                                  .timestamp_format)
+
+        def retrieve_results(metric_config):
+            def now_as_timefield():
+                if metric_config.timestamp_format is 'int':
+                    return time.mktime(datetime.datetime.now().timetuple())
+                else:
+                    return datetime.datetime.now().strftime(
+                        metric_config.timestamp_format)
+
+            def log_error_and_continue(code):
+                known_errors = {
+                    400: 'Bad Request',
+                    401: 'Unauthorized',
+                    403: 'Forbidden',
+                    404: 'Not Found',
+                    409: 'Duplicate',
+                    410: 'Gone',
+                    500: 'Internal Server Error',
+                    501: 'Not Implemented',
+                    503: 'Throttled',
+                    504: 'Gateway Timeout'
+                }
+                try:
+                    self.log.error('Loggly returned error %s: %s'
+                                   % (code, known_errors[code]))
+                except KeyError:
+                    self.log.error('Loggly returned unrecognized error code %s'
+                                   % code)
+                finally:
+                    return []
+
+            search_response = requests.get(
+                metric_config.search_URI,
+                auth=(self.config_bunch.username, self.config_bunch.password))
+            if search_response.status_code is 200:
+                events = requests.get(
+                    metric_config.events_URI % search_response
+                    .json()['rsid']['id'],
+                    auth=(self.config_bunch.username,
+                          self.config_bunch.password))
+
+                if events.status_code is 200:
+                    bare_metrics = [(metric_config.field,
+                                     event[metric_config.timestamp_field],
```

Comment:
So, looking at this, it seems you are trying to shove events into a stats framework. That doesn't seem correct. Usually you would put aggregations into a stats framework, such as the number of events in the last window.